binderhub/main.py

```diff
@@ -7,6 +7,59 @@
 
 from .base import BaseHandler
 
+from tornado.escape import url_escape, url_unescape
+import json
+
+class BinderHandler(BaseHandler):
+    """
+    Base class with utility methods to set and get a json cookie.
+
+    """
+
+
+    def _normalize_cookie(self, cookie):
+        """
+        Make sure the json cookie is normalized to have a know schema:
+        {
+            'known': [list of unique strings]
+            'default': [string found in known section]
+        }
+
+        We could (should ?) be stricter.
+
+        """
+
+        portal_address = self.settings['cannonical_address']
+        default_binders_list = self.settings['default_binders_list']
+        new_cookie = {}
+        new_cookie['known'] = list(sorted(set(cookie.get('known', [portal_address]) + [portal_address] + default_binders_list)))
+        default = cookie.get('default', portal_address)
+        if default not in new_cookie['known']:
+            default = portal_address
+
+        new_cookie['default'] = default
+        return new_cookie
+
+    def set_json_cookie(self, name, value, *args, **kwargs):
+        self.set_cookie(name, url_escape(json.dumps(value)), *args, **kwargs)
+
+    def get_json_cookie(self, name, *args, **kwargs):
+        cookie = self.get_cookie(name, *args, **kwargs)
+        if not cookie:
+            return self._normalize_cookie({})
+        return self._normalize_cookie(json.loads(url_unescape(cookie)))
+
+
+class ExposeHandler(BaseHandler):
+    """Main handler for requests"""
+
+    def get(self):
+        self.render_template(
+            "expose.html",
+            escaped_adresse=url_escape(self.settings['cannonical_address']),
+            google_analytics_code=self.settings['google_analytics_code']
```

Comment:
I think you might need to pass google_analytics_domain too now, see #396

Reply:
Oh, yes.